src/QueryResults.ts

```diff
@@ -0,0 +1,62 @@
+import Promise, { Thenable, Executor } from 'dojo-core/Promise';
+import Task from 'dojo-core/async/Task';
+import {duplicate, mixin} from 'dojo-core/lang';
+import * as dstore from './interfaces';
+
+export interface QueryOptions {
+	totalLength?: number| Promise<number>;
+	response: any;
```

Comment:
Is it really correct for this to be a required property?  I suspect that *only* Request and Rest use it (which happen to be the only things we've considered converting yet).

Granted, if everything in this becomes optional, defining this type becomes more onerous because we need to spell out each permutation (because an object type with all-optional parameters collapses to `any` or `{}` IIRC).